client/nginx.go

```diff
@@ -494,6 +494,60 @@ type HTTPLimitConnections map[string]LimitConnection
 // StreamLimitConnections represents limit connections related stats
 type StreamLimitConnections map[string]LimitConnection
 
+type option func(*NginxClient) error
+
+// WithHTTPClient configures NGINX Plus Client to use customized http.Client.
+//
+//nolint:all
+func WithHTTPClient(httpClient *http.Client) option {
+	return func(nc *NginxClient) error {
+		if httpClient == nil {
+			return errors.New("nil http client")
+		}
+		nc.httpClient = httpClient
+		return nil
+	}
+}
+
+// WithAPIVersion configures NGINX Plus Client to use a specific version of the NGINX Plus API.
+//
+//nolint:all
+func WithAPIVersion(version int) option {
+	return func(nc *NginxClient) error {
+		if !versionSupported(version) {
+			return fmt.Errorf("unsupported API version %v", version)
+		}
+		nc.version = version
+		return nil
+	}
+}
+
+// NewDefaultNginxClient creates the client for the latest NGINX plus version
+// and configured default HTTP timeout set to 10 seconds. User can configure
+// customized http.Client and supported API version by passing options.
+//
+// NewDefaultNginxClient does not perform external http request to determine
+// what NGINX version it talks to. It is responsibility of the caller to make
+// sure the library is used with supported versions.
+func NewDefaultNginxClient(apiEndpoint string, opts ...option) (*NginxClient, error) {
+	if apiEndpoint == "" {
+		return nil, errors.New("api endpoint not specified")
+	}
+	nc := NginxClient{
+		apiEndpoint: apiEndpoint,
+		httpClient: &http.Client{
+			Timeout: 10 * time.Second,
```

Comment:
Is hard-coding a 10 second timeout here really creating a 'default' nginx client? I would have thought leaving the http.Client timeout it its default is more correct. If the user wants to specify 10 seconds then they can do that with WithHTTPClient()